### ISSUE TITLE
adc6 and adc7 configuration fails for devices using atmegax8.h and ad10b_2.h

### ISSUE THREAD

**Issue by gotnone** [duparq/hwa#1]
expansion of the following hwa command to configure adc6 or adc7 fails unexpectedly.

`hwa(configure, adc0, clock, min, trigger, manual, vref, vcc, input, HW_PIN(adc7));`

Running the gcc command with -E allows for the examination of the source after pre-processing.  It appears that `HW_PIN(adc6)` and `HW_PIN(adc7)` are expanding to `pin_adc6` and `pin_adc7`, but the expansion of `HW_ID( HW_PIN(adc6) )` and `HW_ID( HW_PIN(adc7) )` results in 0 for both cases.  For reference `HW_ID( HW_PIN(adc0) )` expands to 111 and `HW_ID( HW_PIN(adc1) )` expands to 112, `HW_ID( HW_PIN(adc5) )` expands to 116.  This causes the comparison for the input (#define _hwa_cfad10b_kinput_1(o,k,v,...)) to fail.

`else if (HW_ID(v)!=0 && HW_ID(v) ==HW_ID( HW_PIN(adc7) ) )`...

looking in `hwa/hwa_macros.h` it can be seen in the comments that `HW_ID(object)` will return 0 if the object does not exist.  Looking at `atmel/avr/devices/atmegax8.h` it appears that `HW_ID( HW_PIN(adc6) )`  should evaluate to 336 and adc7 to 337 if `HW_ID( HW_PIN(adc5) )`, alias `pc5`, expands to 116.  Experimenting with using `adc7` as the input in the ADC configuration instead of HW_PIN(adc7) still fails.  Even though the input adc7 expands to 337, the input comparisons in `#define _hwa_cfad10b_kinput_1(o,k,v,...)`... fail as the comparison of HW_ID(adc7) == HW_ID( HW_PIN(adc7) ), expands to  337 == 0, so the input comparisons fall through to the error condition.

In summary:

- It seems that the expansion of `HW_ID()` is returning 0 when given the arguments of `HW_PIN(adc6)` and `HW_PIN(adc7)`.
- This causes the input parameter evaluation to fail and results in errors during compilation

Attachments:
[test_adc.zip](https://github.com/duparq/hwa/files/2377687/test_adc.zip)
test_adc.zip contains test_adc.c and Makefile

run:
`make BOARD=pro-mini` or any other device that includes `ad10b_2.h` and `atmegax8.h`

**Comment by duparq** [comment]
You've found a bug.

Solution: in the file `atmel/avr/devices/atmegax8.h`, replace 

    #define _hw_def_adc6		_io1a, 336
    #define _hw_def_adc7		_io1a, 337

with

    #define _hw_def_pin_adc6	_io1a, 336
    #define _hw_def_pin_adc7	_io1a, 337

Thank you for your contribution.

**Comment by gotnone** [comment]
Thanks for getting to this so quickly.  I forgot the issue was left open.
